overlord/snapstate/component.go

```diff
@@ -347,6 +347,9 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup ComponentS
 	if fromStore {
 		prepare = st.NewTask("download-component", fmt.Sprintf(i18n.G("Download component %q%s"), compSetup.ComponentName(), revisionStr))
 	} else {
+		// if we're not downloading the component, then we can clear out any
+		// download info that might exist from the calling code
+		compSetup.DownloadInfo = nil
```

Comment:
Is this really necessary? It should be actually nil for local files already, if that is not the case sounds a bit like a bug somewhere.

Reply:
Consider the case where we refresh a snap and the component revisions don't change. In that case, we will still have a `DownloadInfo` on the `ComponentSetup,` since we did check with the store for potential updates.

Since it is valid for doInstallComponent to decide to not download a component, despite the caller providing a `DownloadInfo`, I think that this is needed.

Reply:
0a4266012ba9d3e7883de7d0e4fecad8b84b3512 adds a commit that shows the need for this.

Reply:
but in that case we still want to get a potentially updated pair from the store no? so it's not clear we want this behavior change

Reply:
Ah you're totally right.